modules/numlock.nix

```diff
@@ -0,0 +1,19 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.xsession.numlock;
+in
+  {
+    options = {
+      xsession.numlock.enable = mkEnableOption "Numlock";
+    };
+
+    config = mkIf (cfg.enable) {
+
+      xsession.profileExtra = ''
```

Comment:
Perhaps this would be better as a oneshot systemd service? Something like
```
systemd.user.services.numlockx = {
  Unit = {
    Description = "NumLockX";
    After = [ "graphical-session-pre.target" ];
    PartOf = [ "graphical-session.target" ];
  };

  Service = {
    Type = "oneshot";
    ExecStart = "${pkgs.numlockx}/bin/numlockx";
  };

  Install = {
    WantedBy = [ "graphical-session.target" ];
  };
};
```

Reply:
Good thought.  I was sort of worried about relying on xsession but this seems much less fragile.